### PR TITLE
Y4m nalu compile issues

### PR DESCRIPTION
**Pull request by intelmark**
Several changes to resolve compile warnings on  Linux.

Also changed validateAlphanumeric for y4m testing to better handle y4m header processing

### CODE REVIEW

**Comment by intelmark** [comment]
Let me know what I need to do to get this PR approved.  I don't understand the AppVeyor build failure

**Comment by 1480c1** [comment]
Rebase on master

File: Source/App/EbAppInputy4m.c [hunk 1]
```diff
@@ -43,7 +43,10 @@ int32_t read_y4m_header(EbConfig_t *cfg) {
 
     /* get first line after YUV4MPEG2 */
     fresult = fgets(buffer, sizeof(buffer), ptr_in);
-    assert(fresult != NULL);
+    if (fresult== NULL)
```
Comment (tianjunwork) [inline Source/App/EbAppInputy4m.c]: style.md:
Braces go on the same line for single-line statements, but on a new line for multi-line statements:

File: Source/App/EbAppInputy4m.c [hunk 2]
```diff
@@ -292,7 +295,10 @@ int32_t read_y4m_frame_delimiter(EbConfig_t *cfg) {
         assert(feof(cfg->inputFile));
         return EB_ErrorNone;
     }
-    validateAlphanumeric(bufferY4Mheader);
+    if (!validateAlphanumeric((char*)bufferY4Mheader))
+    {
```
Comment (tianjunwork) [inline Source/App/EbAppInputy4m.c]: do you want to fix all the cases in the file?

File: Source/App/EbAppProcessCmd.c [hunk 3]
```diff
@@ -986,9 +986,9 @@ void SendNaluOnTheFly(
         uint8_t *nextValue = NULL;
 
         if (fgets(line, sizeof(line), config->naluFile) != NULL) {
-            nextValue = EB_STRTOK(line, " ", &context);
+            nextValue = (uint8_t*)EB_STRTOK(line, " ", &context);
```
Comment (tianjunwork) [inline Source/App/EbAppProcessCmd.c]: nextValue is unsigned char

File: Source/App/EbAppInputy4m.c [hunk 4]
```diff
@@ -308,20 +314,21 @@ EB_BOOL check_if_y4m(EbConfig_t *cfg) {
...
         return EB_TRUE; /* YUV4MPEG2 file */
     }
     else {
         if (cfg->inputFile != stdin) {
             fseek(cfg->inputFile, 0, SEEK_SET);
         }
         else {
-            EB_STRNCPY(cfg->y4m_buf, buffer, YUV4MPEG2_IND_SIZE); /* TODO copy 9 bytes read to cfg->y4m_buf*/
+            EB_STRNCPY((char*)cfg->y4m_buf, buffer, YUV4MPEG2_IND_SIZE); /* TODO copy 9 bytes read to cfg->y4m_buf*/
         }
```
Comment (tianjunwork) [inline Source/App/EbAppInputy4m.c]: y4m_buf is unsigned char type. I think it makes more sense that you change buffer to unsigned char.
BTW, raw video data is interpreted to be unsigned char or uint8_t type.

File: Source/App/EbAppInputy4m.c [hunk 5]
```diff
@@ -320,15 +333,15 @@ EB_BOOL check_if_y4m(EbConfig_t *cfg) {
...
+    if (validateAlphanumeric((char*)buffer) && EB_STRCMP((char*)buffer, "YUV4MPEG2") == 0) {
         return EB_TRUE; /* YUV4MPEG2 file */
     }
     else {
         if (cfg->inputFile != stdin) {
             fseek(cfg->inputFile, 0, SEEK_SET);
         }
         else {
-            EB_STRNCPY((char*)cfg->y4m_buf, buffer, YUV4MPEG2_IND_SIZE); /* TODO copy 9 bytes read to cfg->y4m_buf*/
+            EB_STRNCPY((char*)cfg->y4m_buf, (char*)buffer, YUV4MPEG2_IND_SIZE);
```
Comment (tianjunwork) [inline Source/App/EbAppInputy4m.c]: both y4m_buf, buffer are unsigned char type. I don't think you need conversion.

File: Source/App/EbAppInputy4m.c [hunk 6]
```diff
@@ -320,15 +333,15 @@ EB_BOOL check_if_y4m(EbConfig_t *cfg) {
     }
 
     buffer[YUV4MPEG2_IND_SIZE] = 0;
-    if (validateAlphanumeric((char*)buffer) && EB_STRCMP(buffer, "YUV4MPEG2") == 0) {
+    if (validateAlphanumeric((char*)buffer) && EB_STRCMP((char*)buffer, "YUV4MPEG2") == 0) {
```
Comment (tianjunwork) [inline Source/App/EbAppInputy4m.c]: both y4m_buf, buffer are unsigned char type. I don't think you need conversion.

Comment (tianjunwork) [reply Source/App/EbAppInputy4m.c]: validateAlphanumeric(char* buffer)==>validateAlphanumeric(unsigned char* buffer)

File: Source/App/EbAppProcessCmd.c [hunk 7]
```diff
@@ -978,29 +978,29 @@ void SendNaluOnTheFly(
...
             }
             else {
                 return_error = EB_ErrorBadParameter;
             }
             if (return_error == EB_ErrorNone && *context != 0) {
                 nextValue = EB_STRTOK(NULL, "/", &context);
                 if (nextValue !=NULL)
-                    nalType = strtol(nextValue, NULL, 0);
+                    nalType = (uint32_t)strtol((char*)nextValue, NULL, 0);
                 else
```
Comment (tianjunwork) [inline Source/App/EbAppProcessCmd.c]: nextValue is already a char*

File: Source/App/EbAppProcessCmd.c [hunk 8]
```diff
@@ -1010,7 +1010,7 @@ void SendNaluOnTheFly(
             if (return_error == EB_ErrorNone && *context != 0) {
                 nextValue = EB_STRTOK(NULL, " ", &context);
                 if (nextValue != NULL)
-                    payloadType = strtol(nextValue, NULL, 0);
+                    payloadType = (uint32_t)strtol((char*)nextValue, NULL, 0);
                 else
```
Comment (tianjunwork) [inline Source/App/EbAppProcessCmd.c]: nextValue is already a char*

**Pull request merged**